From 2ba39284ca2cac761b70bdc3839114944dca0f19 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 29 Dec 2022 07:52:10 +0200
Subject: [PATCH] [SW-113135] gaudi3: use bootinfo mem region for bootinfo set

Since new SRAM BOOTINFO partition is added, this patch updates
bootinfo set function to make use of SRAM BOOTINFO region instead
SRAM GP region.

Change-Id: Ib2b2683a66685ad895f14224556c84c7aea0df66
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index ae357b3..6306382 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -131,8 +131,8 @@
 	uintptr_t mem_base, mem_base_end;
 	u32 i, actual_size, ret = 0;
 
-	if (preboot_get_mem_region(MEM_GP, &mem_region_desc)) {
-		preboot_puts("boot_info_set: failed to get mem region MEM_GP\n");
+	if (preboot_get_mem_region(MEM_BOOT_INFO, &mem_region_desc)) {
+		preboot_puts("boot_info_set: failed to get mem region MEM_BOOT_INFO\n");
 		return 1;
 	}
 
@@ -146,7 +146,7 @@
 			continue;
 		}
 
-		/* copy boot info memory into GP mem region */
+		/* copy boot info memory into BOOT_INFO mem region */
 		if (mem_base + desc->size > mem_base_end) {
 			preboot_puts("boot_info_set: not enough space for bootinfo\n");
 			ret = 1;
